src/BlueSCSI_disk.cpp

```diff
@@ -2113,6 +2149,16 @@ void scsiDiskPoll()
             image_config_t &img = *(image_config_t*)scsiDev.target->cfg;
             checkDiskGeometryDivisible(img);
         }
+
+        // Check for Inquiry command to reinsert CD-ROMs on boot
+        if (command == 0x12)
+        {
+            image_config_t &img = *(image_config_t*)scsiDev.target->cfg;
+            if (img.deviceType == S2S_CFG_OPTICAL && img.reinsert_on_inquiry)
```

Comment:
Smaller perf penalty would be to change reinsert on inquiry first in the if block